packages/nodes-base/nodes/Spotify/Spotify.node.ts

```diff
@@ -0,0 +1,548 @@
+
+		if( resource === 'player' ) {
+			if(['pause'].includes(operation)) {
+				requestMethod = 'PUT';
+
+				endpoint = `/me/${resource}/${operation}`;
+			} else if(['recently-played', 'currently-playing'].includes(operation)) {
+				requestMethod = 'GET';
+
+				endpoint = `/me/${resource}/${operation}`;
+			} else if(['next', 'previous'].includes(operation)) {
+				requestMethod = 'POST';
+
+				endpoint = `/me/${resource}/${operation}`;
+			} else if(['play'].includes(operation)) {
+				requestMethod = 'PUT';
+
+				const id = this.getNodeParameter('resourceID', 0) as string;
+
+				body.context_uri = `${id}`;
+
+				endpoint = `/me/${resource}/${operation}`;
+			} else if(['queue'].includes(operation)) {
+				requestMethod = 'POST';
+
+				const id = this.getNodeParameter('trackID', 0) as string;
+
+				endpoint = `/me/${resource}/${operation}?uri=${id}`;
+			}
+		} else if( resource == 'albums') {
+			const uri = this.getNodeParameter('albumID', 0) as string;
+
+			const id = uri.replace('spotify:album:', '');
+
+			requestMethod = 'GET';
+
+			if(['albums'].includes(operation)) {
+				endpoint = `/${resource}/${id}`;
+			} else if(['tracks'].includes(operation)) {
+				endpoint = `/${resource}/${id}/${operation}`;
+			}
+		} else if( resource == 'artists') {
+			const uri = this.getNodeParameter('artistID', 0) as string;
+
+			const id = uri.replace('spotify:artist:', '');
+
+			requestMethod = 'GET';
+
+			if(['albums','related-artists'].includes(operation)) {
+				endpoint = `/${resource}/${id}/${operation}`
+			} else if(['top-tracks'].includes(operation)){
+				const country = this.getNodeParameter('country', 0) as string;
+
+				endpoint = `/${resource}/${id}/top-tracks?country=${country}`
+			} else if(['artist'].includes(operation)) {
+				endpoint = `/${resource}/${id}`
+			}
+		} else if( resource == 'playlists') {
+			if(['delete'].includes(operation)) {
+				requestMethod = 'DELETE';
+
+				const uri = this.getNodeParameter('playlistID', 0) as string;
+
+				const id = uri.replace('spotify:playlist:', '');
+
+				const trackId = this.getNodeParameter('trackID', 0) as string;
+
+				body.tracks = [
+					{
+					  "uri": `${trackId}`,
+					  "positions": [
+						0
+					  ]
+					}
+				]
+
+				endpoint = `/${resource}/${id}/tracks`;
+			} else if(['user-playlists'].includes(operation)) {
+				requestMethod = 'GET';
+
+				endpoint = `/me/${resource}`;
+			} else if(['get'].includes(operation)) {
+				requestMethod = 'GET';
+
+				const uri = this.getNodeParameter('playlistID', 0) as string;
+
+				const id = uri.replace('spotify:playlist:', '');
+
+				endpoint = `/${resource}/${id}`;
+			} else if(['add'].includes(operation)) {
+				requestMethod = 'POST';
+
+				const uri = this.getNodeParameter('playlistID', 0) as string;
+
+				const id = uri.replace('spotify:playlist:', '');
+
+				const trackId = this.getNodeParameter('trackID', 0) as string;
+
+				endpoint = `/${resource}/${id}/tracks?uris=${trackId}`;
+			}
```

Comment:
Gosh, this is just one of those things that's just a *little* too complex to be an elegant map.

Reply:
That is a theme of the codebase! 😞